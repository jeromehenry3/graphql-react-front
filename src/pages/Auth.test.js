import React from 'react';
import ReactDOM from 'react-dom';
import AuthPage from './Auth';
import renderer from 'react-test-renderer';
import ReactTestUtils from 'react-dom/test-utils';
import { cleanup, render, simulate } from '@testing-library/react';
import expectExport from 'expect';




// it('renders without crashing', () => {
//   const div = document.createElement('div');
//   render(<AuthPage />, div);
//   ReactDOM.unmountComponentAtNode(div);
// });

// it('should render a <h1> containing : Erreur 404 : la page que vous recherchez n\'existe pas.', () => {
//     const div = document.createElement('div');
//     render(<AuthPage />, div);
//     expect(div).toHaveTextContent('Erreur 404 : la page que vous recherchez n\'existe pas.')
// })

test('should render <AuthPage /> as defined', () => {
  const component = renderer.create(
    <AuthPage />
  );
  let tree = component.toJSON();
  expectExport(tree).toMatchSnapshot();

  // Changement de vue
    const instance = component.root.findByProps({id: "signup"});
    ReactTestUtils.Simulate.click(instance)



  // expect(component).
});
