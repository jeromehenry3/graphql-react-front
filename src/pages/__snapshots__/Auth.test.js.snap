// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render <AuthPage /> as defined 1`] = `
<form
  className="ui form login-form"
  onSubmit={[Function]}
>
  <div
    className="field"
  >
    <label>
      email
    </label>
    <div
      className="ui input"
    >
      <input
        onChange={[Function]}
        placeholder="email"
        type="text"
        value=""
      />
    </div>
  </div>
  <div
    className="field"
  >
    <label>
      mot de passe
    </label>
    <div
      className="ui input"
    >
      <input
        onChange={[Function]}
        placeholder="mot de passe"
        type="password"
        value=""
      />
    </div>
  </div>
  <div
    className="field"
  >
    <button
      className="ui button"
      id="signup"
      onClick={[Function]}
    >
      Cr√©er un compte
    </button>
  </div>
  <div
    className="field"
  >
    <button
      className="ui button"
      onClick={[Function]}
      type="submit"
    >
      Se connecter
    </button>
  </div>
</form>
`;
